### PR TITLE
Convert tests from rspec to pester

### PR DESCRIPTION
**Pull request by matt-richardson**
We've had a lot of trouble fighting with ruby - mainly around rubygems. It's just not fun.
Combine that with unfamiliarity and a "why are we using ruby for tests when the rest of the code is in powershell" disconnect, it makes a lot of sense to use Pester as our test framework instead.

When we merge this, we should mark the [octopus-serverspec-extensions](https://github.com/OctopusDeploy/octopus-serverspec-extensions) repo as archived.

### CODE REVIEW

**Review by MissedTheMark — APPROVED** [review]
Overall concept is a good one.  Definitely a better testing setup than what we were doing before.

File: E2ETests/Spec/TestHelpers/Get-TentacleDetails.ps1 [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        $OctopusApiKey,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $InstanceName,
+        [Parameter(Mandatory=$false)]
+        [string]
+        $SpaceId
+    )
+
+    [bool]$exists = Test-Path "c:\program files\Octopus Deploy\Tentacle\Tentacle.exe"
```
Comment (cailyoung) [inline E2ETests/Spec/TestHelpers/Get-TentacleDetails.ps1:19]: I assume we only run on Windows and this path is sufficiently well-known.

File: E2ETests/Spec/server_scenario_01_install.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    #dsc overall status
+    it "should be able to get dsc configuration" {
+        $ProgressPreference = "SilentlyContinue"
+        $state = ""
+        do {
+            $state = (Get-DscLocalConfigurationManager).LCMState
+            write-verbose "LCM state is $state"
+            Start-Sleep -Seconds 2
+        } while ($state -ne "Idle")
```
Comment (cailyoung) [inline E2ETests/Spec/server_scenario_01_install.Tests.ps1:133]: Is there a timeout on these?

**Review by cailyoung — APPROVED** [review]
LGTM. I'm not familiar enough with all the context to know if you've reimplemented the assertions properly, but the shape of all of this looks right.

I note that the increased test count is due to _more_ tests being reported as tests to TeamCity where previously they were effectively reports within logging.

You've showed me a diff of the old and new test lists (after _extensive_ data cleaning!) and there don't seem to be any missing in the Pester suite.

**Pull request merged**